### PR TITLE
kickoff v18 reef

### PR DESCRIPTION
**Pull request by rzarzynski**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
jenkins test windows

File: src/include/ceph_features.h [hunk 1]
```diff
@@ -184,7 +184,8 @@ DEFINE_CEPH_FEATURE(60, 1, OSD_RECOVERY_DELETES) // *do not share this bit*
 DEFINE_CEPH_FEATURE(61, 1, CEPHX_V2)         // 4.19, *do not share this bit*
 
 DEFINE_CEPH_FEATURE(62, 1, RESERVED)           // do not use; used as a sentinel
-DEFINE_CEPH_FEATURE_DEPRECATED(63, 1, RESERVED_BROKEN, LUMINOUS) // client-facing
+DEFINE_CEPH_FEATURE_RETIRED(63, 1, RESERVED_BROKEN, LUMINOUS, QUINCY) // client-facing
+// available
```
Comment (idryomov) [inline src/include/ceph_features.h:188]: > X-3 == R-3 == Octopus

I don't think `RESERVED_BROKEN` was tied to Octopus or the "X-3" rule.   But deciding that it has been long enough for `OSD_SNAPMAPPER` workaround to go away entirely is fine too :-)

File: src/mon/Monitor.h [hunk 2]
```diff
@@ -1115,6 +1115,7 @@ class Monitor : public Dispatcher,
 #define CEPH_MON_FEATURE_INCOMPAT_OCTOPUS CompatSet::Feature(12, "octopus ondisk layout")
 #define CEPH_MON_FEATURE_INCOMPAT_PACIFIC CompatSet::Feature(13, "pacific ondisk layout")
 #define CEPH_MON_FEATURE_INCOMPAT_QUINCY CompatSet::Feature(14, "quincy ondisk layout")
+#define CEPH_MON_FEATURE_INCOMPAT_REEF CompatSet::Feature(15, "quincy ondisk layout")
```
Comment (idryomov) [inline src/mon/Monitor.h:1118]:
```suggestion
#define CEPH_MON_FEATURE_INCOMPAT_REEF CompatSet::Feature(15, "reef ondisk layout")
```

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -3523,28 +3523,28 @@ bool OSDMonitor::preprocess_boot(MonOpRequestRef op)
...
   if (HAVE_FEATURE(m->osd_features, SERVER_QUINCY) &&
       osdmap.require_osd_release < ceph_release_t::octopus) {
     mon.clog->info() << "disallowing boot of quincy+ OSD "
 		      << m->get_orig_source_inst()
 		      << " because require_osd_release < octopus";
     goto ignore;
   }
+  if (HAVE_FEATURE(m->osd_features, SERVER_REEF) &&
+      osdmap.require_osd_release < ceph_release_t::pacific) {
+    mon.clog->info() << "disallowing boot of quincy+ OSD "
```
Comment (idryomov) [inline src/mon/OSDMonitor.cc:3543]:
```suggestion
    mon.clog->info() << "disallowing boot of reef+ OSD "
```

File: doc/dev/release-checklists.rst [hunk 4]
```diff
@@ -17,7 +17,7 @@ Versions and tags
 
 - [x] Update CMakeLists.txt VERSION (right at the top to X.0.0)
 - [x] Update src/ceph_release with the new release name, number, and type ('dev')
-- [x] Initial tag vX.0.0 (so that we can distinguish from (and sort
+- [ ] Initial tag vX.0.0 (so that we can distinguish from (and sort
```
[the remark this answers is not in the document]
Comment (rzarzynski) [reply doc/dev/release-checklists.rst:20]: I would prefer to create the tag after the review process is complete.

File: src/include/ceph_features.h (hunk 1 above)
Comment (rzarzynski) [reply src/include/ceph_features.h:188]: If we can retire this bit, let's go for it.

Just to the record the meaning of the macros' params:

```cpp
// this bit is ignored but still advertised by release *when*
#define DEFINE_CEPH_FEATURE_DEPRECATED(bit, incarnation, name, when) \
        const static uint64_t DEPRECATED_CEPH_FEATURE_##name = (1ULL<<bit); \
        const static uint64_t DEPRECATED_CEPH_FEATUREMASK_##name =              \
                (1ULL<<bit | CEPH_FEATURE_INCARNATION_##incarnation);

// this bit is ignored by release *unused* and not advertised by
// release *unadvertised*
#define DEFINE_CEPH_FEATURE_RETIRED(bit, inc, name, unused, unadvertised)
```

**Comment by rzarzynski** [comment]
The make check bot failed early:

```
  C++20 support requires a minimum Clang version of 12.
```

**Comment by rzarzynski** [comment]
jenkins test make check

**Comment by cbodley** [comment]
cc @tchaikov motr still causing 'make check' failures:
```
/ceph-pull-requests/src/rgw/rgw_sal.cc
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_sal.cc:33:
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_sal_motr.h:25:
In file included from /usr/include/motr/motr/client.h:1875:
/usr/include/motr/cas/cas.h:38:10: fatal error: 'fop/wire.h' file not found
#include "fop/wire.h"           /* m0_fop_mod_rep */
         ^~~~~~~~~~~~
1 error generated.
```
i've opened https://github.com/ceph/ceph/pull/47564 to disable WITH_RADOSGW_MOTR in make check until we stabilize our CI

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test api

**Comment by idryomov** [comment]
"make check" failure pertains to this PR:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/monmaptool/feature-set-unset-list.t: failed
--- /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/monmaptool/feature-set-unset-list.t
+++ /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/monmaptool/feature-set-unset-list.t.err
@@ -12,21 +12,21 @@
       required:   [none]
   
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
   MONMAP FEATURES:
       persistent: [none]
       optional:   [none]
       required:   [none]
   
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
   monmap:persistent:[none]
   monmap:optional:[none]
   monmap:required:[none]
-  available:supported:[kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-  available:persistent:[kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+  available:supported:[kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+  available:persistent:[kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
 
   $ monmaptool --feature-set foo /tmp/test.monmap.1234
   unknown features name 'foo' or unable to parse value: Expected option value to be integer, got 'foo'
@@ -45,8 +45,8 @@
       required:   [kraken(1),octopus(32),unknown(4096)]
   
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
 
   $ monmaptool --feature-unset 32 --optional --feature-list /tmp/test.monmap.1234
   monmaptool: monmap file /tmp/test.monmap.1234
@@ -56,8 +56,8 @@
       required:   [kraken(1),octopus(32),unknown(4096)]
   
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
   monmaptool: writing epoch 0 to /tmp/test.monmap.1234 (1 monitors)
 
   $ monmaptool --feature-unset 32 --persistent --feature-unset 4096 --optional --feature-list /tmp/test.monmap.1234
@@ -68,8 +68,8 @@
       required:   [kraken(1)]
   
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
   monmaptool: writing epoch 0 to /tmp/test.monmap.1234 (1 monitors)
 
   $ monmaptool --feature-unset kraken --feature-list /tmp/test.monmap.1234
@@ -80,8 +80,8 @@
       required:   [none]
   
   AVAILABLE FEATURES:
-      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
-      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256)]
+      supported:  [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
+      persistent: [kraken(1),luminous(2),mimic(4),osdmap-prune(8),nautilus(16),octopus(32),pacific(64),elector-pinging(128),quincy(256),reef(512)]
   monmaptool: writing epoch 0 to /tmp/test.monmap.1234 (1 monitors)
 
   $ rm /tmp/test.monmap.1234
```

**Comment by rzarzynski** [comment]
Added `reef` to `feature-set-unset-list.t` and extended the checklist to cover this step as well.

**Comment by rzarzynski** [comment]
```
The following tests FAILED:
	 32 - run-rbd-unit-tests-61.sh (Timeout)
	235 - unittest-seastar-messenger (Failed)
```

**Comment by rzarzynski** [comment]
jenkins test make check

**Comment by rzarzynski** [comment]
```
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find zbd (missing: ZBD_LIBRARIES ZBD_INCLUDE_DIR)
```

**Comment by rzarzynski** [comment]
jenkins test make check

File: src/mon/MonCommands.h [hunk 5]
```diff
@@ -839,7 +839,7 @@ COMMAND("osd unset "
 	"notieragent|nosnaptrim",
 	"unset <key>", "osd", "rw")
 COMMAND("osd require-osd-release "\
-	"name=release,type=CephChoices,strings=octopus|pacific|quincy "
+	"name=release,type=CephChoices,strings=octopus|pacific|quincy|reef "
```
Comment (jdurgin) [inline src/mon/MonCommands.h:842]: can remove octopus

File: src/mon/OSDMonitor.cc [hunk 6]
```diff
@@ -11595,27 +11595,14 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
...
-      if ((!HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_OCTOPUS))
-           && !sure) {
-	ss << "not all up OSDs have CEPH_FEATURE_SERVER_OCTOPUS feature";
-	err = -EPERM;
-	goto reply;
-      }
-    } else if (rel == ceph_release_t::pacific) {
+    if (rel == ceph_release_t::pacific) {
       if (!mon.monmap->get_required_features().contains_all(
 	    ceph::features::mon::FEATURE_PACIFIC)) {
```
Comment (jdurgin) [inline src/mon/OSDMonitor.cc:11607]: there are a couple checks for SERVER_OCTOPUS that we can get rid of now - around using MRemoveSnaps

File: doc/dev/release-checklists.rst [hunk 7]
```diff
@@ -17,7 +17,7 @@ Versions and tags
...
   after) the backported (X-1).2.Z versions.
```
Comment (jdurgin) [inline doc/dev/release-checklists.rst:21]: should add OSDMap::pending_require_osd_release() update

**Review by jdurgin — APPROVED** [review]
a few minor comments, looking good! could handle cleaning up older checks / deprecating feature bits in another pr

File: src/mon/MonCommands.h (hunk 5 above)
Comment (rzarzynski) [reply src/mon/MonCommands.h:842]: Fixed.

File: doc/dev/release-checklists.rst (hunk 7 above)
Comment (rzarzynski) [reply doc/dev/release-checklists.rst:21]: Thanks for pointing this out! Fixed `OSDMap::pending_require_osd_release()` and added the item to the checklist.

**Comment by tchaikov** [comment]
jenkins test api

**Review by neha-ojha — APPROVED** [review]
Thanks @rzarzynski!
@yuriw let's start running this PR through all the suites

**Comment by ljflores** [comment]
jenkins test api

**Review by anthonyeleven — APPROVED** [review]
Not a lot of docness here, so lgtm

**Comment by adk3798** [comment]
jenkins test make check

**Review by adk3798 — APPROVED** [review]

File: qa/standalone/mon/misc.sh [hunk 8]
```diff
@@ -265,6 +270,8 @@ function TEST_mon_features() {
     jq_success "$jqinput" "$jqfilter" || return 1
```
Comment (neha-ojha) [inline qa/standalone/mon/misc.sh:270]: http://pulpito.front.sepia.ceph.com/yuriw-2022-09-14_16:47:07-rados-wip-yuri4-testing-2022-09-14-0617-distro-default-smithi/7032918/  is related. We need to update this now - this was done is later commit for Quincy https://github.com/ceph/ceph/pull/39187

**Comment by rzarzynski** [comment]
Added to the release checklist a section for changes to `ceph-build`.

**Comment by rzarzynski** [comment]
Resolved a merge conflict with acd6d4efea0aa8b64ff2f196c6c7647b734c9d90 on `qa/suites/upgrade/octopus-x/rgw-multisite/realm.yaml` – we both were removing this file.

**Review by cbodley — APPROVED** [review]

**Comment by rzarzynski** [comment]
```
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/release-checklists.rst:25: ERROR: Unexpected indentation.
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/release-checklists.rst:26: WARNING: Block quote ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
```

Fixing. No other things than just the `release-checklist.rst` will be affected.

**Comment by rzarzynski** [comment]
Absolutely no other changes except the doc's indentation:

```
commit 65d911cc9bdc393a546ba553e3c9189a8f7e32c1
Author: Radoslaw Zarzynski <rzarzynski@redhat.com>
Date:   Tue Oct 4 00:27:12 2022 +0200

    fixup
    
    Signed-off-by: Radosław Zarzyński <rzarzyns@redhat.com>

diff --git a/doc/dev/release-checklists.rst b/doc/dev/release-checklists.rst
index bb4bc5bd3c2..4b7817b1dcc 100644
--- a/doc/dev/release-checklists.rst
+++ b/doc/dev/release-checklists.rst
@@ -21,13 +21,12 @@ Versions and tags
       after) the backported (X-1).2.Z versions.
 
 ### Notes on tagging
-* Tags must be annonated as CMake determines `CEPH_GIT_NICE_VER` by calling
-  `git describe --always`.
+* Tags must be annonated as CMake determines `CEPH_GIT_NICE_VER` by
+calling `git describe --always`.
 * vX.0.0 are special ones in the sense they are pushed manually (unlike v.X.2.n
-  which are handled by Jenkins).
+which are handled by Jenkins).
 * vX.0.0 should point to a commit before the first one in a kickoff branch.
 
-
 Define release names and constants
```

**Comment by idryomov** [comment]
Bumped librbd version and added respective items to the checklist.

**Comment by ljflores** [comment]
jenkins retest this please

**Comment by ljflores** [comment]
@rzarzynski I only found one related failure. But I think you fixed it in a subsequent commit since it passed in the rerun.
http://pulpito.front.sepia.ceph.com/yuriw-2022-09-29_16:40:30-rados-wip-all-kickoff-r-distro-default-smithi/7047967/
```
2022-09-29T20:41:39.878 INFO:tasks.workunit.client.0.smithi164.stderr://home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2321: jq_success:  echo '
2022-09-29T20:41:39.879 INFO:tasks.workunit.client.0.smithi164.stderr:{"election_epoch":14,"quorum":[0,1,2],"quorum_names":["a","b","c"],"quorum_leader_name":"a","quorum_age":4,"features":{"quorum_con":"4540138322906710015","quorum_mon":["kraken","luminous","mimic","osdmap-prune","nautilus","octopus","pacific","elector-pinging","quincy","reef"]},"monmap":{"epoch":3,"fsid":"2a5cab5f-e7e2-46a5-ac28-ea1b6b85400e","modified":"2022-09-29T20:41:29.707699Z","created":"2022-09-29T20:40:55.456287Z","min_mon_release":18,"min_mon_release_name":"reef","election_strategy":1,"disallowed_leaders: ":"","stretch_mode":false,"tiebreaker_mon":"","features":{"persistent":["kraken","luminous","mimic","osdmap-prune","nautilus","octopus","pacific","elector-pinging","quincy","reef"],"optional":[]},"mons":[{"rank":0,"name":"a","public_addrs":{"addrvec":[{"type":"v2","addr":"127.0.0.1:7127","nonce":0}]},"addr":"127.0.0.1:7127/0","public_addr":"127.0.0.1:7127/0","priority":0,"weight":0,"crush_location":"{}"},{"rank":1,"name":"b","public_addrs":{"addrvec":[{"type":"v2","addr":"127.0.0.1:7128","nonce":0}]},"addr":"127.0.0.1:7128/0","public_addr":"127.0.0.1:7128/0","priority":0,"weight":0,"crush_location":"{}"},{"rank":2,"name":"c","public_addrs":{"addrvec":[{"type":"v2","addr":"127.0.0.1:7129","nonce":0}]},"addr":"127.0.0.1:7129/0","public_addr":"127.0.0.1:7129/0","priority":0,"weight":0,"crush_location":"{}"}]}}'
2022-09-29T20:41:39.879 INFO:tasks.workunit.client.0.smithi164.stderr://home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2321: jq_success:  jq '.monmap.features.persistent | length == 9'
2022-09-29T20:41:39.906 INFO:tasks.workunit.client.0.smithi164.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2321: jq_success:  ret=false
2022-09-29T20:41:39.907 INFO:tasks.workunit.client.0.smithi164.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2322: jq_success:  [[ false == \t\r\u\e ]]
2022-09-29T20:41:39.907 INFO:tasks.workunit.client.0.smithi164.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2324: jq_success:  [[ -n "" ]]
2022-09-29T20:41:39.907 INFO:tasks.workunit.client.0.smithi164.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2325: jq_success:  [[ false == \"\" ]]
2022-09-29T20:41:39.907 INFO:tasks.workunit.client.0.smithi164.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/ceph-helpers.sh:2329: jq_success:  return 1
2022-09-29T20:41:39.908 INFO:tasks.workunit.client.0.smithi164.stderr:/home/ubuntu/cephtest/clone.client.0/qa/standalone/mon/misc.sh:270: TEST_mon_features:  return 1
```

Rados suite review:
http://pulpito.front.sepia.ceph.com/?branch=wip-all-kickoff-r

Failures, unrelated:
    1. https://tracker.ceph.com/issues/57386
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/57731
    4. https://tracker.ceph.com/issues/57311
    5. https://tracker.ceph.com/issues/50042
    6. https://tracker.ceph.com/issues/57546

Details:
    1. cephadm/test_dashboard_e2e.sh: Expected to find content: '/^foo$/' within the selector: 'cd-modal .badge' but never did - Ceph - Mgr - Dashboard
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    3. Problem: package container-selinux conflicts with udica < 0.2.6-1 provided by udica-0.2.4-1 - Infrastructure
    4. rook: ensure CRDs are installed first - Ceph - Orchestrator
    5. rados/test.sh: api_watch_notify failures - Ceph - RADOS
    6. rados/thrash-erasure-code: wait_for_recovery timeout due to "active+clean+remapped+laggy" pgs - Ceph - RADOS

**Review by ljflores — APPROVED** [review]

**Review by guits — APPROVED** [review]

**Comment by travisn** [comment]
Rook daily tests against Ceph main are green since this is merged and Rook can detect the Reef version, thanks!

**Comment by cbodley** [comment]
yikes, i hadn't noticed that this PR inadvertently removed rgw from the upgrade suites entirely. are the upgrade suites cephadm-only now?

**Comment by cbodley** [comment]
okay, i see this just removed all octopus-x upgrade suites, and we never made any "no-cephadm" upgrade suites after that

**Pull request merged**